src/GafferScene/RenderController.cpp

```diff
@@ -823,20 +972,55 @@ class RenderController::SceneGraph
 			m_dirtyComponents &= ~components;
 		}
 
+		M44f fullTransform( float time ) const
+		{
+			if( m_fullTransform.empty() )
+			{
+				return M44f();
+			}
+			if( !m_transformTimesOutput )
+			{
+				return m_fullTransform[0];
+			}
+
+			vector<float>::const_iterator t1 = lower_bound( m_transformTimesOutput->begin(), m_transformTimesOutput->end(), time );
+			if( t1 == m_transformTimesOutput->begin() || *t1 == time )
+			{
+				return m_fullTransform[t1 - m_transformTimesOutput->begin()];
+			}
+			else
+			{
+				vector<float>::const_iterator t0 = t1 - 1;
+				const float l = lerpfactor( time, *t0, *t1 );
+				const M44f &s0 = m_fullTransform[t0 - m_transformTimesOutput->begin()];
+				const M44f &s1 = m_fullTransform[t1 - m_transformTimesOutput->begin()];
+				M44f result;
+				LinearInterpolator<M44f>()( s0, s1, l, result );
+				return result;
+			}
+		}
+
 		IECore::InternedString m_name;
 
 		const SceneGraph *m_parent;
 
 		IECore::MurmurHash m_objectHash;
 		ObjectInterfaceHandle m_objectInterface;
+		std::vector<float> m_deformationTimes;
 
 		IECore::MurmurHash m_attributesHash;
 		IECore::CompoundObjectPtr m_fullAttributes;
 		IECoreScenePreview::Renderer::AttributesInterfacePtr m_attributesInterface;
 		IECore::MurmurHash m_lightLinksHash;
 
 		IECore::MurmurHash m_transformHash;
-		Imath::M44f m_fullTransform;
+		std::vector<Imath::M44f> m_fullTransform;
```

Comment:
Might be worth using `small_vector` here and for `m_deformationTimes`? Typically we're only expecting two floats...

Reply:
Sounds kinda reasonable, but we're not saving anything if we have to convert to vector, right?  Are you proposing changing the interface to IECoreScenePreview::Renderer::ObjectInterface::transform?  Might be reasonable, but probably one for another time?

Reply:
Good point - yep, we can leave for another time.